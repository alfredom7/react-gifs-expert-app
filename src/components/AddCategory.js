import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState(''); // ''

    const handleInputChange = ( e ) => {
        setInputValue( e.target.value );
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if ( inputValue.trim().length > 2 ) {
            setCategories( cats => [ inputValue ] );
            setInputValue('');
        }

    }

    return (
        <form onSubmit={ handleSubmit }>
            <p>{ inputValue }</p>
            <div class="input-group mb-3">
            <input 
                type="text" 
                value={ inputValue } 
                onChange={ handleInputChange } 
                className="form-control" 
                placeholder="Write for search..." 
                aria-label="Write for search..." 
                aria-describedby="button-addon2"
            />
            <button onClick={ handleSubmit } className="btn btn-success" type="button" id="button-addon2">Search</button>
            </div>
        </form>
    )
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
