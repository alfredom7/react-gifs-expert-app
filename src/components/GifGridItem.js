import React from 'react'
import PropTypes from 'prop-types'

export const GifGridItem = ({ title, url }) => {

    return (
        <div className='col-md-4 col-lg-3 col-xl-2 pb-3'>
            <div className="mycard card text-white bg-dark animate__animated animate__fadeIn shadow-sm rounded-3">
            <img src={ url } alt={ title } className="card-img-top img-grid"/>
            <div className="card-body text-center">
                <p className="card-title img-my-title text-capitalize text-truncate">{ title }</p>
                <a href="#" className="btn btn-success btn-sm">Select</a>
            </div>
            </div>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}