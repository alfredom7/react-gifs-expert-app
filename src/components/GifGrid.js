import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types'

export const GifGrid = ({ category }) => {

    const { data:images, loading } = useFetchGifs( category );

    return (
        <>
            <h6 className="animate__animated animate__fadeIn pb-2 resultsfor">Results for: { category } </h6>

            { loading && <p className="animate__animated animate__flash">Loading</p> }

            <div className="card-grid row">
                
                {
                    images.map( img => (
                        <GifGridItem 
                            key={ img.id }
                            { ...img }
                        />
                    ))
                }
            
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
