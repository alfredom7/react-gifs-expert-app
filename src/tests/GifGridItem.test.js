import React from 'react';
import {shallow} from 'enzyme'
import { GifGridItem } from '../components/GifGridItem';

const title = 'My test title';
const url = 'myapp.alfredo.com'
const wrapper = shallow(<GifGridItem title={title} url={url}/>);

describe('Pruebas en <GifGridItem/>', () => {
  
  test('Debe mostrar el componente correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe tener un parrafo con el titulo', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  });

  test('Debe tener la imagen igual al url y alt de los props', () => {
    const img = wrapper.find('img');
    expect(img.props().src).toBe(url)
  });

  test('debe tener animate__fadeIn', () => {
    const div = wrapper.find('div.mycard');
    const className = div.prop('className');
    expect(className.includes('animate__fadeIn')).toBe(true)
  });


});