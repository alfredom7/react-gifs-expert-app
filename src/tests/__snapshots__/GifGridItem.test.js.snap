// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> Debe mostrar el componente correctamente 1`] = `
<div
  className="col-md-4 col-lg-3 col-xl-2 pb-3"
>
  <div
    className="mycard card text-white bg-dark animate__animated animate__fadeIn shadow-sm rounded-3"
  >
    <img
      alt="My test title"
      className="card-img-top img-grid"
      src="myapp.alfredo.com"
    />
    <div
      className="card-body text-center"
    >
      <p
        className="card-title img-my-title text-capitalize text-truncate"
      >
        My test title
      </p>
      <a
        className="btn btn-success btn-sm"
        href="#"
      >
        Select
      </a>
    </div>
  </div>
</div>
`;
